app/models/study.rb

```diff
@@ -1132,6 +1132,22 @@ def unique_genes
     Gene.where(study_id: self.id, :study_file_id.in => self.expression_matrix_files.map(&:id)).pluck(:name).uniq
   end
 
+  # Get species scientific name for a searched gene.  E.g. PTEN -> Homo sapiens.
+  def infer_species(searched_gene)
```

Comment:
It seems to me that this method is doing more work than it needs to.  If the goal is simply to return a scientific name for a searched gene, we can probably do that easier.  The `Gene` model has an instance method called [`taxon`](https://github.com/broadinstitute/single_cell_portal_core/blob/development/app/models/gene.rb#L56) that is a getter for the expression matrix `Taxon` instance.  We could rewrite this method as a one-liner that would return an array of all possible species:

```
def infer_species(gene_name)
  Gene.where(study_id: self.id, :study_file_id.in => self.expression_matrix_files.pluck(:id), name: gene_name).map {|gene| gene.taxon.try(:scientific_name)}.uniq
end
```

Reply:
Done [here](https://github.com/broadinstitute/single_cell_portal_core/blob/1731b32e9d0f2a7fbd6d89c177b5eebc95ca0641/app/models/study.rb#L1135), including some readability and consistency refinements.